lib/intl.ts

```diff
@@ -118,7 +118,10 @@ function DateTimeFormatImpl(
   const hasOptions = typeof optionsParam !== 'undefined';
   const options = hasOptions ? ObjectAssign({}, optionsParam) : {};
   // TODO: remove type assertion after Temporal types land in TS lib types
-  const original = new IntlDateTimeFormat(locale, options as globalThis.Intl.DateTimeFormatOptions);
+  const original = new IntlDateTimeFormat(
```

Comment:
Is it correct that once https://github.com/microsoft/TypeScript/pull/52996 lands and we update our TS version in use we will be able to stop casting the locale param? If so, would you mind adding a TODO here linking against that PR? Thanks.

Reply:
Correct - I've added a TODO but referencing the issue not the PR, in case that PR doesn't get merged for some reason